distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -35,6 +40,20 @@ get_format callback =
                 @Tail_Call reader (idx + 1)
     reader 0
 
+## PRIVATE
+format_widget : Single_Choice
+format_widget =
+    all_types = [Auto_Detect] + format_types
+    make_ctor type_obj =
+        type_name = Meta.get_qualified_type_name type_obj
+        ctor_name = if type_obj==JSON_Format || (type_name.ends_with "_Format" . not) then "" else
```

Comment:
This logic is a bit convoluted. What is its purpose?
I assume it is meant to distinguish formats that are represented by a 'singleton type' vs ones that need to have a constructor, is that right? Would appreciate then naming this condition somehow:
```suggestion
        is_singleton_type = type_obj==JSON_Format || (type_name.ends_with "_Format" . not)
        ctor_name = if is_singleton_type then "" else
```